learningequality/studio
# Changing languages

## Description

**Draft PR because we don't want to merge it yet, but definitely would appreciate review as the remaining work is not going to change the core of the implementation.**

I moved over and tweaked some code from Kolibri to get our language switching working properly on the Studio side.

Also included latest strings from Crowdin updating FR and AR.

**Remaining Work**

- ~Spanish is not working - I will look into this ASAP.~
- ~Will need to have new strings that were added auto-translated on Crowdin - want to get those translated and pulled in if I can before merging.~ (DONE)
- Aim to replace `console.error()` with an error message Sentry will report.
- Will make a follow-up issue about creating an error message whenever there is an issue w/ switching languages. Currently we just `console.error()` but we should notify the user - just can't add strings right now.

#### Issue Addressed (if applicable)

Fixes #2585 

## Steps to Test

- Check that you can change the language on the Sign In page at the bottom there.
- Check that you can navigate without having any issues with language reverting to English or a previous selection. Navigate between sections of the site like Channel Editing, Channel Listing, Setting, Profile.
- Check that changing your language in the dropdown menu in the AppBar has the same proper effect.


## Implementation Notes (optional)

#### At a high level, how did you implement this?

I went into Kolibri and just deKolibrified it, made some tweaks to the language handling and made it work. 

Key thing to note is that the default Django `i18n/setlang` just didn't work outright - so I took Kolibri's version of it and used that as the View for that path instead.
